### DISCUSSION TITLE
Whisper doesn't translate in non-english anymore

### DISCUSSION THREAD

**Discussion by PawBroon** [openai/whisper#649] [Q&A] [answered]
It might have been a slip in the crack or a happy accident which got unknowingly patched in a latter Commit, but when transcribing an English video with language XYZ (Most European languages and a few Asian ones) and --transcribe, it generated a PERFECTLY translated files.
As defined in XYZ in lieu of --language en. 

And it translated while being asked to transcribe.
I've used this and tested it with almost all languages in the tokenizer, and it worked way better than what Google Translate ever produced, when being fed the English transcript of Whispered YouTube video.

I don't know which commit killed it, as I've done a handful of Git Checkout ID . sadly to no avail.
Just so you know, you had a perfectly functional translating machine, from English to more than a dozen foreign languages, but now it only translate into English.
Worked only when asking for a foreign languages --transcribe of an English speaking file, tagged as being fr/es/de/it and so on.
Medium model, using Cuda...

Anyone had a similar experience, or better yet, is still able to trick Whisper into translating to non-officially supported languages?

**Comment by jongwook** [comment, marked as answer]
Translating into languages other than English wasn't a part of the training objective, so that behavior may not be always reliable. I suspect the new `"large-v2"` model added in the latest commit [4179ed2](https://github.com/openai/whisper/commit/4179ed2475cc84cba66868b516232ef1b74dacdf) might be the causing the change you are experiencing.

If that's the case, you can specify `"large-v1"` to use the original large model.

**Reply by PawBroon** [reply]
I've always thought it was a byproduct of an oversight or something, since that feature was never discussed anywhere.
To be fair, I've only stumbled upon it accidentally, as I've typed --language fr out of habit, and the --transcribe did a perfect French translation of an English speaking audio input. 

What's weird is that I don't use Large, as I'm sadly using a RTX2060 with not enough VRAM to load this model.
But while I've suspected this, using a previous commit checkout didn't solve this "issue".
And I know it's NOT an issue per-se, as it was never intended to work.
But since it did up to days ago, I just wanted to let you know that Whisper had this up its sleeve, as an added bonus you never knew you had.

Happy accidents, just like when you comment your Code to let everyone knows that it works, but you can't say why.
So steer clear of it, pretty please...
😆

**Reply by yoboitel** [reply]
Hey @PawBroon ! Have you manage to get it working ? Should we use the previous model to translate ? I'm also interested by whisper because i'm looking to translate english videos to french transcribed text so hope it still works :)

**Reply by jere76** [reply]
I tried v1 for the same purpose of yours (es > fr), without success. Not sure what would be the syntax to use though.
whisper --model large-v1 --task translate --language fr /file.mp3 translates in English just as v2

**Reply by nicolasraj** [reply]
@jere76 this works for me
whisper audio.mp3 --model medium --task transcribe --language French

**Reply by jere76** [reply]
hey @nicolasraj, thanks a lot, it works ! I was using translate instead of transcribe... I guess I was not really well awake when I read this first post...

**Reply by pepeluisavi** [reply]
this whisper audio.mp3 --model medium --task transcribe --language French works perfectly, only bad deal is that without gpu delays eons to translate, and you may need to pay for premium gpus after some time, that or manually translate your files, that may be faster if you know english-your language

**Reply by CorAnna** [reply]
@nicolasraj Sorry for the perhaps trivial question, do you know how I should do if I want to integrate the translation into any language from a python file?
And should I install which version with pip?

Thank you in advance

**Reply by RobertChalmers** [reply]
Is Audio.mp3 English, and the output is French?

"this whisper audio.mp3 --model medium --task transcribe --language French works perfectly"

**Reply by RobertChalmers** [reply]
So Whisper does translate from English to French for example now?

What is the command then, because I can't make it work.

**Comment by mayeaux** [comment]
If the translations were as accurate as you say, it would be great to be able to use Whisper for translation. I hope OpenAI releases an open-source translation model, the cost of translation keeps going up it's an industry that could use some disruption in a big way. With Whisper the cost of producing high quality transcriptions is driven down significantly but to translate an hour of content say with Google Translate (fixed rate of $20/million characters) costs about $0.80/h of transcribed content, which ends up being way higher than the actual cost of transcription itself, while translation (based on my amateur guess) should be much less computationally expensive.

**Reply by RobertChalmers** [reply]
Whisper is translating for me from English to French and Spanish. Using big model, and a '5' wait time. 
Nice

**Comment by PawBroon** [comment]
Indeed.
Way back when, Whisper did DeepL level translations in languages it wasn't
documented to support.

Le dim. 9 juil. 2023 à 10:18, ShanghaiTimes ***@***.***> a
écrit :

**Comment by PawBroon** [comment]
Not *NOW*, months ago before a new release which broke the functionality.
I've raised it and they replied that as it was never an intended feature,
it's not broken and work as it should.
No more translation outside of English.

Le dim. 9 juil. 2023 à 11:24, ShanghaiTimes ***@***.***> a
écrit :

**Reply by RobertChalmers** [reply]
. However, it does sometimes translate. Especially short pieces.
I can easily translate a 1 minute long English to French piece but not a 15-minute one. 

For example. (Except for the last line for some reason?)

% whisper /Users/robertchalmers/Music/PlayHT/TheWitchesStoryIntroduction.wav --model medium --language fr --task transcribe
/usr/local/lib/python3.9/site-packages/whisper/transcribe.py:114: UserWarning: FP16 is not supported on CPU; using FP32 instead
  warnings.warn("FP16 is not supported on CPU; using FP32 instead")
[00:00.000 --> 00:04.600]  Lillie était une jeune fille ordinaire qui vivait dans une petite ville sur les étoiles de la ville.
[00:04.600 --> 00:08.760]  Elle aimait explorer le paysage et avait une certaine fierté pour son jardin,
[00:08.760 --> 00:12.600]  où elle a passé de nombreuses heures heureuses à lutter contre ses fleurs et des herbes.
[00:12.600 --> 00:18.600]  Un jour, en jouant dans le jardin, Lillie s'est plongée sur un petit portier de pierre caché entre les arbres.
[00:18.600 --> 00:23.800]  Intrigue, elle s'est allumée et sentait une énergie étrange éminente des pierres.
[00:23.800 --> 00:29.600]  Sans hésitation, Lillie s'est allumée dans le portier et a découvert un monde qui n'était pas comme celui qu'elle avait vu avant.
[00:29.600 --> 00:34.000]  Little did she know, this was just the beginning of her magical journey as a young witch.

**Reply by RobertChalmers** [reply]
A second run through using the "small" model.

 whisper /Users/robertchalmers/Music/PlayHT/TheWitchesStoryIntroduction.wav --model small --language fr --task transcribe
/usr/local/lib/python3.9/site-packages/whisper/__init__.py:63: UserWarning: /Users/robertchalmers/.cache/whisper/small.pt exists, but the SHA256 checksum does not match; re-downloading the file
  warnings.warn(
100%|███████████████████████████████████████| 461M/461M [04:02<00:00, 1.99MiB/s]
/usr/local/lib/python3.9/site-packages/whisper/transcribe.py:114: UserWarning: FP16 is not supported on CPU; using FP32 instead
  warnings.warn("FP16 is not supported on CPU; using FP32 instead")
[00:00.000 --> 00:04.640]  Lili était une jeune ordinaire qui vivait dans un petit village sur les outskirts de la ville.
[00:04.640 --> 00:08.640]  Elle aimait explorer le côté du pays et avait une faune particulière pour son jardin,
[00:08.640 --> 00:12.000]  où elle a pris beaucoup de bonheur, tendant à ses fleurs et les herbes.
[00:12.640 --> 00:16.080]  Un jour, en jouant dans le jardin, Lili s'est arrêté sur un petit gâteau d'une petite
[00:16.080 --> 00:21.840]  peau dans les bouches. Intrigue, elle s'est arrêtée plus loin et a senti une énergie étrange
[00:21.840 --> 00:27.200]  emanant de la peau. Sans hésitation, Lili s'est arrêté au gâteau et a trouvé un monde
[00:27.280 --> 00:30.880]  unlike qu'elle n'avait jamais vu avant. Elle ne connait rien.
[00:30.880 --> 00:33.840]  C'était juste le début de sa voyage magique comme une jeune fille.

**Comment by PawBroon** [comment]
It's nice to know it's working somewhat, insyead of not at all.
But the results are not too French, Mesium yielded way better results.
Since I only have a 2060 with 6gb VRAM, I've never been able to run Large.

Le dim. 9 juil. 2023 à 12:23, ShanghaiTimes ***@***.***> a
écrit :

**Reply by RobertChalmers** [reply]
Yes, only somewhat... I'm on an M2 Mac Mini, with 26GB Ram. The 'large' model only gives me English on a long file of 14 minutes. I'm trying it again now but it takes ages to download the model.
And now so far so good. This is a big audio file. 271MB. Duration 15.43

whisper "/Users/robertchalmers/Movies/FCPX Movies/Removing and Replacing Selected Colours From An Image.wav" --model large --language French                   
100%|█████████████████████████████████████| 2.87G/2.87G [25:06<00:00, 2.05MiB/s]
/usr/local/lib/python3.9/site-packages/whisper/transcribe.py:114: UserWarning: FP16 is not supported on CPU; using FP32 instead
  warnings.warn("FP16 is not supported on CPU; using FP32 instead")
[00:00.000 --> 00:12.000]  Salut les gars, on va voir comment changer les couleurs d'une image.
[00:12.000 --> 00:15.640]  Commençons par celui-ci, quelque chose de stupidement dégueulasse.
[00:15.640 --> 00:39.120]  Vous voyez beaucoup de choses sur les réseaux sociaux, et les gens se demandent comment
[00:39.120 --> 00:52.120]  de changer les couleurs d'une image, et c'est ce que je vais faire, j'ai choisi
[00:52.120 --> 00:59.120]  l'application Affinity Designer 2, qui fonctionne exactement comme sur l'iPad et sur le PC.
[01:00.120 --> 01:11.120]  Vous pouvez voir que c'est bloqué, alors faisons un duplicate, et on peut éteindre cette couche.
[01:11.120 --> 01:16.120]  Parce que c'est dégueulasse, cette couche est éteinte.
[01:17.120 --> 01:21.120]  Passons à l'écran, on est dans le mode des pixels.
[01:21.120 --> 01:23.120]  Ça peut se déclencher dans un instant.
[01:23.120 --> 01:25.120]  Non, ça ne va pas se déclencher.
[01:25.120 --> 01:26.120]  Bien joué.
[01:26.120 --> 01:38.120]  OK, il y a un pinceau d'éraser, et on a un pinceau assez grand, 64 pixels, ça va le faire.
[01:38.120 --> 01:45.120]  Et on peut éteindre tout ce rouge, tout ça, tout ça.

**Comment by RobertChalmers** [comment]
So it seems very short files translate pretty well. Long ones don't.
Again, a short file. And I wish I could stop the warning about FP16 not being available.

ackages/whisper/transcribe.py:114: UserWarning: FP16 is not supported on CPU; using FP32 instead
  warnings.warn("FP16 is not supported on CPU; using FP32 instead")
[00:00.000 --> 00:04.400]  Après sa première aventure, Lily ne pouvait pas attendre pour retourner au monde enchanté.
[00:04.400 --> 00:09.280]  Elle a spent tous les moments de lire des livres sur la magie et de pratiquer ses spels.
[00:09.280 --> 00:13.600]  Finalement, le jour est venu pour qu'elle puisse s'éteindre et s'éteindre à travers la gate de la paix de la paix encore une fois.
robertchalmers@M2Mini PlayHT %

**Comment by RobertChalmers** [comment]
Ok, it seems to depend on the model.
whisper "/Users/robertchalmers/Movies/FCPX Movies/Removing and Replacing Selected Colours From An Image.wav" --model large --language French 

This works, all the way through.
This is a 15:42 wav file. The Mac is an M2 Mac Miini, 24GB RAM.
The French translation took 1 hour 30 minutes.

The Spanish version has dropped into English after 5 sentences.

**Reply by G444-4** [reply]
Hi @ShanghaiTimes, what was your command for this task because I’m trying too to translate an audio to a text in French using whisper but it keeps doing it in English even if I specify that I want the large-v1 model (by the way I’m running my code on google collab if that makes any difference).

**Reply by gandolfi974** [reply]
hi, have you found a way to translate a video or audio file into french? thanks

**Comment by PawBroon** [comment]
90 minutes for 15' of source file?!
I'd have bet the M2 would outperform my RTX.
Last time I've tried with a similar input, I thought it was awfully long,
but lasted way less than 60 minutes of a 2060.
I guess there are no substitute to CUDA cores.

Le dim. 9 juil. 2023 à 14:43, ShanghaiTimes ***@***.***> a
écrit :

**Comment by PawBroon** [comment]
Good to know, as I hope to switch back to Mac.

Le dim. 9 juil. 2023 à 17:00, ShanghaiTimes ***@***.***> a
écrit :

**Comment by phineas-pta** [comment]
for the time being if u want to use on apple silicon better use https://github.com/ggerganov/whisper.cpp

**Comment by RobertChalmers** [comment]
HiAs I mention in my post, simply using a 5 wait time I think.I haven’t used it in a while now. I gave up.-----Robert ***@***.*** 7 Jan 2024, at 7:47 am, G444-4 ***@***.***> wrote:﻿
Hi @ShanghaiTimes, what was your command for this task because I’m trying too to translate an audio to a text in French using whisper but it keeps doing it in English even if I specify that I want the large-v1 model (by the way I’m running my code on google collab if that makes any difference).

**Comment by thewh1teagle** [comment]
Is there some news about this feature?
I would like to translate **into** other languages than English. Currently translating works only into English.

**Reply by G444-4** [reply]
Well, I personally gave up on this method. And used Whisper to transcribe and translate into English and then I other models  from that I found on Huggingface or google translate to translate into the other language (results weren't as good as I would hoped to tho, but it might be enough for some). Since GPT4o is supposed to be able to translate by design I think it would be possible to use it from the API (but I think tou would have to pay for it). If you're in one of the listed countries maybe try to use Gemini 🤷‍♂️

**Reply by G444-4** [reply]
Hi I've just seen that a company called Gladia Ia could help you

**Comment by 3bagorion33** [comment]
This just worked
`faster-whisper-xxl.exe --model large-v2 --language German "Video.mp4"`
The one and a half hour Russian video has been translated into German.

**Comment by qrevol** [comment]
Hello,
I'm on an M1 MacBookPro, 16GB Mem. Using the large-v2 model for "transcription" worked like a charm to translate a 10-min. video from English into Portuguese. It took about 30min or less.

**Comment by pr0mila** [comment]
I’ve created a code-switched language dataset for fine-tuning Whisper, including audio data along with CSV and Parquet files, which I’ve stored on Hugging Face. After preparing the dataset, I fine-tuned the model for translation. You can explore the entire end-to-end project in my repo. Here’s the link to check it out: https://github.com/pr0mila/MediBeng-Whisper-Tiny
